hyp3_gamma/__main__.py

```diff
@@ -112,6 +112,17 @@ def rtc():
             upload_file_to_s3(product_file, args.bucket, args.bucket_prefix)
 
 
+def valid_range_float(x):
+    try:
+        x = float(x)
+    except ValueError:
+        raise ArgumentTypeError("%r not a floating-point literal" % (x,))
+
+    if x < 0.0 or x > 1.0:
+        raise ArgumentTypeError("%r not in range [0.0, 1.0]" % (x,))
+    return x
```

Comment:
```suggestion
def valid_range_float(x):
    x = float(x)
    if 0.0 <= x <= 1.0:
        return x
    raise ValueError(f'{x} not in range [0.0, 1.0]')


```

For string formatting, you should always prefer f-strings over `.format` strings over `%`. 

You also do not need to raise an `ArgumentTypeError` here, argparse will do that for you automatically when specifying this function as a type.